app/services/migration/migrators/contract.rb

```diff
@@ -30,6 +32,18 @@ def ecf_contracts
       def dependencies
         %i[course statement]
       end
+
+      def number_of_workers
```

Comment:
Could we just wrap the contract creation in a transaction to avoid the race conditions? alternatively I think instead of overriding the constant you could just do something like this I think?:

```
def number_of_workers
  1
end

def records_per_worker
  record_count
end
```

Reply:
Hi @ethax-ross tried that yesterday.. but it breaks some shared specs..